src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -16,14 +16,40 @@ internal partial class ClrThreadPool
         public static readonly ClrThreadPool ThreadPoolInstance = new ClrThreadPool();
 #pragma warning restore IDE1006 // Naming Styles
 
+        private const short MaxPossibleThreadCount = short.MaxValue;
+
+        private static short GetThreadCountConfig(string configName)
+        {
+            var value = AppContext.GetData(configName);
+            ushort numThreads = 0;
+            switch (value)
```

Comment:
Would be good to use the same helper between HillClimbing and here for this part